core/src/main/scala/Manifest.scala

```diff
@@ -69,8 +69,8 @@ object Manifest {
   }
 
   final case class Environment(
-    cpu: Option[Double] = None,
-    memory: Option[Double] = None,
+    cpu: Option[(Double, Double)] = None,
```

Comment:
I'm not wild about this tuple - can we either type alias `Double` to something semantically meaningful, or provide an ADT?

Reply:
Leaning towards doing an ADT since that seems like the more typeful way to do things. I'm thinking something like:

```scala
sealed abstract class ResourceSpec[A] extends Product with Serializable {
  def fold[B](unspecified: B)(limitOnly: A => B)(bounded: (A, A) => B): B = ...
}

object ResourceSpec {
  // Hide the constructors since we want to make smart constructors for 'bounded'
  private class Unspecified[A]() extends ResourceSpec[A]
  private class LimitOnly[A](limit: A) extends ResourceSpec[A]
  private class Bounded[A](request: A, limit: A) extends ResourceSpec[A]

  def unspecified[A]: ResourceSpec[A] = new Unspecified[A]()

  def limitOnly[A](limit: A)(implicit A: Numeric[A]): Option[ResourceSpec[A]] =
    if (A.gteq(limit, A.zero)) Some(new LimitOnly(limit))
    else None

  def bounded[A](request: A, limit: A)(implicit A: Numeric[A]): Option[ResourceSpec[A]] =
    if (A.gteg(limit, request) && A.gteq(limit, A.zero)) Some(new Bounded(request, limit))
    else None
}
```

This would fit with us validating everything in the parser and from then on anything that sees `ResourceSpec` can be assured and patmat on if a resource is unspecified, limit only, or bounded, with appropriate checks. WDYT?

Reply:
done!